src/rust/engine/src/core.rs

```diff
@@ -71,12 +71,19 @@ pub struct Function(pub Key);
 /// Wraps a type id for use as a key in HashMaps and sets.
 ///
 #[repr(C)]
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy)]
 pub struct Key {
   id: Id,
   type_id: TypeId,
 }
 
+// This is consumed in user-facing error messages, such as for --glob-expansion-failure=error.
```

Comment:
We definitely need to do an overhaul of error messages at some point. But if you're using this in exactly one place, I'd recommend just explicitly calling `externs::key_to_str` there (the user won't care that it is a `Key` wrapped around their value... just the value).

Reply:
I reverted this -- see below!